### PR TITLE
Adds title and description frontmatter to docs Reference section

### PR DESCRIPTION
**Pull request by dthyresson**
This PR adds title and description frontmatter to clean up the reference section docs to avoid displaying just the first line of the page content. Also, used for SEO.

image

### CODE REVIEW

**Comment by jtoar** [comment]
@dthyresson Thanks for getting started on this! Not sure why this isn't building; I'll have a look.

File: docs/docs/cli-commands.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+---
+title: Command Line Interface
+description: A comprehensive reference of the Redwood CLI
+slug: cli
```
Comment (jtoar) [inline docs/docs/cli-commands.md:4]:
```suggestion
```

File: docs/docs/toast-notifications.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+---
+title: Toast Notifications
+description: Render toast notifications with built-in components
+slug: toast
```
Comment (jtoar) [inline docs/docs/toast-notifications.md:4]:
```suggestion
```

**Comment by jtoar** [comment]
@dthyresson found out what happened: the way we have it setup right now, docusaurus throws if there's broken links and refuses to build, and we changed a few slugs (cli, toast-notifications) that some of the docs were linking to.

Do you think we could keep this PR scoped to just the descriptions on the "docs/index" page by foregoing the `slug` field for now? Some of the slugs could be better for sure. But this PR would have to include a whole bunch of other changes to make it all work

**Comment by dthyresson** [comment]
> docusaurus throws if there's broken links and refuses to build,

@jtoar Oh, I didn't see that behavior when running the dev server (as I did not build).

I'll review again to fix slugs (or remove) and test a build before committing. Thanks.

**Comment by jtoar** [comment]
@dthyresson I didn't see it locally either actually! Only showed up on netlify 🤔

**Comment by dthyresson** [comment]
FYI - If I set a slug and it breaks a docs link:

```
Exhaustive list of all broken links found:

- On source page path = /docs/seo:
   -> linking to /docs/prerender
```

**Comment by dthyresson** [comment]
@jtoar Looks like you have fixes all the links and it currently builds; do I need to fixe any others -- or is the PR ok? We'll keep the slugs as they were and defined in the sidebar set.

File: docs/docs/webpack-configuration.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+---
+title: Webpack Configuration
+description: If you have to configure webpack, here's how.
```
Comment (jtoar) [inline docs/docs/webpack-configuration.md:3]: Love it! 😆

**Comment by jtoar** [comment]
@dthyresson I'm reading each one now; I'll make sure the slugs are the same as I do

**Comment by jtoar** [comment]
@dthyresson I had to shorten some so that they fit on the "docs/index" page and don't cut off into an ellipsis. Everything looks good to go to me now!

**Pull request merged**